src/jira/client/jira-api-client.ts

```diff
@@ -0,0 +1,494 @@
+			operationType: options?.operationType || "NORMAL"
+		};
+
+		this.logger.info({ ...extractDeploymentDataForLoggingPurpose(data, this.logger) }, "Sending deployments payload to jira.");
+		const response: AxiosResponse = await this.axios.post("/rest/deployments/0.1/bulk", payload);
+
+		if (
+			response.data?.rejectedDeployments?.length ||
+			response.data?.unknownIssueKeys?.length ||
+			response.data?.unknownAssociations?.length
+		) {
+			this.logger.warn({
+				acceptedDeployments: response.data?.acceptedDeployments,
+				rejectedDeployments: response.data?.rejectedDeployments,
+				unknownIssueKeys: response.data?.unknownIssueKeys,
+				unknownAssociations: response.data?.unknownAssociations,
+				options,
+				...getDeploymentDebugInfo(data)
+			}, "Jira API rejected deployment!");
+		} else {
+			this.logger.info({
+				acceptedDeployments: response.data?.acceptedDeployments,
+				options,
+				...getDeploymentDebugInfo(data)
+			}, "Jira API accepted deployment!");
+		}
+
+		return {
+			status: response.status,
+			rejectedDeployments: response.data?.rejectedDeployments
+		};
+	}
+
+	// TODO TEST
+	async submitRemoteLinks(data, options?: JiraSubmitOptions) {
+		// Note: RemoteLinks doesn't have an issueKey field and takes in associations instead
+		updateIssueKeyAssociationValuesFor(data.remoteLinks, uniq);
+		if (!withinIssueKeyAssociationsLimit(data.remoteLinks)) {
+			updateIssueKeyAssociationValuesFor(data.remoteLinks, truncate);
+			const subscription = await Subscription.getSingleInstallation(this.jiraHost, this.gitHubInstallationId, this.gitHubAppId);
+			await subscription?.update({ syncWarning: issueKeyLimitWarning });
+		}
+		const payload = {
+			remoteLinks: data.remoteLinks,
+			properties: {
+				gitHubInstallationId: this.gitHubInstallationId
+			},
+			preventTransitions: options?.preventTransitions || false,
+			operationType: options?.operationType || "NORMAL"
+		};
+		this.logger.info("Sending remoteLinks payload to jira.");
+		await this.axios.post("/rest/remotelinks/1.0/bulk", payload);
+	}
+
+	// TODO TEST
+	async submitVulnerabilities(data: JiraVulnerabilityBulkSubmitData, options?: JiraSubmitOptions): Promise<AxiosResponse> {
+		const payload = {
+			vulnerabilities: data.vulnerabilities,
+			properties: {
+				gitHubInstallationId: this.gitHubInstallationId
+			},
+			operationType: options?.operationType || "NORMAL"
+		};
+		this.logger.info("Sending vulnerabilities payload to jira.");
+		return await this.axios.post("/rest/security/1.0/bulk", payload);
+	}
+}
+
+// TODO MOVE TO new jira-client-commit-helper.ts
+const deduplicateCommits = (commits: JiraCommit[] = []): JiraCommit[] => {
+	const uniqueCommits = commits.reduce((accumulator: JiraCommit[], currentCommit: JiraCommit) => {
+		if (!accumulator.some((commit) => commit.id === currentCommit.id)) {
+			accumulator.push(currentCommit);
+		}
+		return accumulator;
+	}, []);
+	return uniqueCommits;
+};
+
+// TODO MOVE TO new jira-client-commit-helper.ts
```

Comment:
More TODO i was getting concious of the PR size, deduplicateCommits and batchedBulkUpdate will be moved to commit-helper file